lib/jobs/ucs-discovery.js

```diff
@@ -186,15 +199,69 @@ function UcsDiscoveryJobFactory(
         });
     };
 
+    UcsDiscoveryJob.prototype.updateRelations = function(nodeId, relations) {
```

Comment:
The relations are always mutual relations, such as "encloses" vs. "enclosedBy". If if you change the node which has "encloses" relation, the its peer node need be updated as well.